src/test/scala/chiselTests/Direction.scala

```diff
@@ -379,4 +381,50 @@ class DirectionSpec extends ChiselPropSpec with Matchers with Utils {
     assert(emitted.contains("io.monitor.valid <= io.driver.valid"))
     assert(emitted.contains("io.monitor.ready <= io.driver.ready"))
   }
+  property("Bugfix: marking Vec fields with mixed directionality as Output clears inner directions") {
+    class Decoupled extends Bundle {
+      val bits = UInt(3.W)
+      val valid = Bool()
+      val ready = Flipped(Bool())
+    }
+    class DecoupledAndMonitor extends Bundle {
+      val driver = Output(Vec(1, new Decoupled()))
+    }
+    class MyModule extends RawModule {
+      val io = IO(new DecoupledAndMonitor())
```

Comment:
can you test both Input and Output case?

Reply:
also can you test a case where it is not within a Bundle:

val io = IO(Input(Vec (1, new Decoupled())

Reply:
Yup happy to :)